src/pybind/mgr/diskprediction/agent/metrics/ceph_osd.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+
+import socket
+
+from . import MetricsAgent
+from ...common.db import DB_API
+from ...models.metrics.dp import Ceph_OSD
+
+
+class CephOSD_Agent(MetricsAgent):

```

Comment:
The CephOSD and CephMon are mostly duplicated, and they both pull *all* cluster metrics out and filter by type.  Can we combine to halve the amount of processing that's needed in building that big dict and then throwing it away?

Reply:
@liewegas Thanks, I will improve it.